lib/Server.js

```diff
@@ -87,9 +87,8 @@ class Server {
       throw new Error("'filename' option must be set in lazy mode.");
     }
 
-    if (options.spdy || options.http2) {
-      options.spdy = true;
-      options.http2 = true;
+    if (options.http2 && !options.https) {
+      throw new Error("'https' option must be enabled to use HTTP/2");
```

Comment:
I was wrong, it is poor DX, better set `https: true` in this case, no one browsers don't support `http2` without `ssl` so, we can generate `certificate` sefely